src/domain/payments/payment-flow-builder.ts

```diff
@@ -117,8 +117,13 @@ export const LightningPaymentFlowBuilder = <S extends WalletCurrency>(
 const LPFBWithInvoice = <S extends WalletCurrency>(
   state: LPFBWithInvoiceState,
 ): LPFBWithInvoice<S> | LPFBWithError => {
-  const withSenderWallet = (senderWallet: WalletDescriptor<S>) => {
-    const { id: senderWalletId, currency: senderWalletCurrency } = senderWallet
+  const withSenderWallet = (senderWallet: Wallet) => {
+    const {
+      id: senderWalletId,
+      accountId: senderAccountId,
+    }: { id: WalletId; accountId: AccountId } = senderWallet
+    const senderWalletCurrency = senderWallet.currency as S
+
```

Comment:
Is this really necessary? We're loosing some type safety with the cast and also we still don't need the whole wallet right? Perhaps `accountId` needs to be part of WalletDescriptor. Would like to dig into this and see if there isn't a better implementation.

Reply:
Yea we don't need the whole wallet. I changed the type exactly just to get accountId and was wondering the same about whether we can include it as part of `WalletDescriptor`

Reply:
Yeah lets include it then.

Reply:
Ok I started doing this and ended up getting all the way to our `WalletInvoice` and `LedgerTransaction` persistence types that depend on `WalletDescriptor` as part of their translations. This might be a refactor that's better left to a separate PR, unless we'd really like to make this distinction here.

Reply:
I tackled this again and was able to work around having to change those persistence types by injecting the already available `accountId` in [one instance](https://github.com/GaloyMoney/galoy/pull/1682/commits/fbaf87546b6d6ed3b93f3e9ab3278acb4c82135a) and by fetching the `accountId` [in another](https://github.com/GaloyMoney/galoy/pull/1682/commits/7b183f9e479365bd7f717f85dfb97a4ff4ce166d).

Changeset: https://github.com/GaloyMoney/galoy/compare/b5cc88b3d3e39eb9ced9e01a5b87b6b6f68054bb...eb3faca25ca20a08f72ec4d32b0e06bd32aa4f41